Problems with Skylake support: missing hbwmalloc.h header

Thanks for the beautiful work. I have have however a problem in compiling it, if I try to specifically enable Intel Skylake optimizations, in attempt to replicate the comparison against MKL from: [Intel simple SGEMM/DGEMM benchmark](https://software.intel.com/en-us/articles/benchmarking-gemm-with-intel-mkl-and-blis-on-intel-processors)

`./configure --enable-threading=openmp skx`

`./configure skx
make clean ; make V=1`

... fails, due to not finding the header hbwmalloc.h (from [memkind](https://github.com/memkind/memkind), I understand?)

```
make V=1
./build/flatten-headers.py -c -v1 ./frame/include/blis.h include/skx/blis.h "./include" "./ ./config/skx/ ./kernels/skx/ ./kernels/skx/3/ ./frame/0/ ./frame/0/copysc/ ./frame/1/ ./frame/1d/ ./frame/1f/ ./frame/1m/ ./frame/1m/packm/ ./frame/1m/scalm/ ./frame/1m/unpackm/ ./frame/2/ ./frame/2/gemv/ ./frame/2/ger/ ./frame/2/hemv/ ./frame/2/her/ ./frame/2/her2/ ./frame/2/symv/ ./frame/2/syr/ ./frame/2/syr2/ ./frame/2/trmv/ ./frame/2/trsv/ ./frame/3/ ./frame/3/gemm/ ./frame/3/hemm/ ./frame/3/her2k/ ./frame/3/herk/ ./frame/3/symm/ ./frame/3/syr2k/ ./frame/3/syrk/ ./frame/3/trmm/ ./frame/3/trmm3/ ./frame/3/trsm/ ./frame/base/ ./frame/base/check/ ./frame/base/noopt/ ./frame/compat/ ./frame/compat/cblas/ ./frame/compat/cblas/f77_sub/ ./frame/compat/cblas/src/ ./frame/compat/check/ ./frame/compat/f2c/ ./frame/compat/f2c/util/ ./frame/include/ ./frame/include/level0/ ./frame/include/level0/1e/ ./frame/include/level0/1m/ ./frame/include/level0/1r/ ./frame/include/level0/io/ ./frame/include/level0/ri/ ./frame/include/level0/ri3/ ./frame/include/level0/rih/ ./frame/include/level0/ro/ ./frame/include/level0/rpi/ ./frame/ind/ ./frame/ind/cntx/ ./frame/ind/misc/ ./frame/ind/oapi/ ./frame/ind/tapi/ ./frame/ind/ukernels/ ./frame/thread/ ./frame/util/"
.....................................................................................................................................................................................................................................................................................................................................................................................................................................................
gcc -O3 -mavx512f -mavx512dq -mavx512bw -mavx512vl -mfpmath=sse -march=skylake-avx512 -Wall -Wno-unused-function -Wfatal-errors -fPIC -std=c99 -D_POSIX_C_SOURCE=200112L -fopenmp -I./include/skx -I./frame/3/ -I./frame/ind/ukernels/ -I./frame/1m/ -I./frame/1f/ -I./frame/1/ -I./frame/include -DBLIS_VERSION_STRING=\"0.2.2-137\" -c config/skx/bli_cntx_init_skx.c -o obj/skx/config/skx/bli_cntx_init_skx.o
In file included from config/skx/bli_cntx_init_skx.c:35:0:
./include/skx/blis.h:16156:34: fatal error: hbwmalloc.h: No such file or directory
 #include <hbwmalloc.h> // skipped
                                  ^
compilation terminated.
Makefile:452: recipe for target 'obj/skx/config/skx/bli_cntx_init_skx.o' failed
make: *** [obj/skx/config/skx/bli_cntx_init_skx.o] Error 1
```

Could you please indicate this dependence in future revisions of 'configure' ?

Thanks!

--- a/config/skx/bli_family_skx.h
+++ b/config/skx/bli_family_skx.h
@@ -50,21 +50,10 @@
 #define BLIS_SIMD_SIZE                   64
 #define BLIS_SIMD_NUM_REGISTERS          32
 
-#ifdef BLIS_NO_HBWMALLOC
-
 #include <stdlib.h>
 
 #define BLIS_MALLOC_POOL malloc
 #define BLIS_FREE_POOL free
-
-#else
-
-#include <hbwmalloc.h>
-
-#define BLIS_MALLOC_POOL hbw_malloc
-#define BLIS_FREE_POOL hbw_free
-
-#endif
 
 
 #if 0

--- a/config/skx/make_defs.mk
+++ b/config/skx/make_defs.mk
@@ -60,13 +60,9 @@ CDBGFLAGS      := -g
 endif
 
 ifeq ($(DEBUG_TYPE),noopt)
-COPTFLAGS      := -O0 -DBLIS_NO_HBWMALLOC
+COPTFLAGS      := -O0
 else
 COPTFLAGS      := -O3
-endif
-
-ifeq ($(DEBUG_TYPE),sde)
-CPPROCFLAGS    += -DBLIS_NO_HBWMALLOC
 endif
 
 CKOPTFLAGS     := $(COPTFLAGS)
@@ -100,11 +96,7 @@ ARFLAGS        := cr
 LINKER         := $(CC)
 SOFLAGS        := -shared
 
-ifneq ($(DEBUG_TYPE),sde)
-LDFLAGS        := -lmemkind
-else
 LDFLAGS        :=
-endif
 
 ifneq ($(CC_VENDOR),icc)
 LDFLAGS        += -lm

